### PR TITLE
CRM-16707 Make token usage the responsibality of the form

### PR DESCRIPTION
**Pull request by mallezie**

### CODE REVIEW

**Comment by colemanw** [comment]
This looks like a great refactor :)

**Comment by mallezie** [comment]
These test failures look very unrelated to me.

**Comment by mallezie** [comment]
The same test failures are showing in (for example) https://github.com/civicrm/civicrm-core/pull/6077
So definitely unrelated.

**Comment by monishdeb** [comment]
Jenkin, test this please

File: CRM/Mailing/BAO/Mailing.php [hunk 1]
```diff
@@ -2637,22 +2637,8 @@ public function getReturnProperties() {
...
-    }
-    elseif ($className == 'CRM_Admin_Form_ScheduleReminders') {
-      $tokens = array_merge(CRM_Core_SelectValues::activityTokens(), $tokens);
-      $tokens = array_merge(CRM_Core_SelectValues::eventTokens(), $tokens);
-      $tokens = array_merge(CRM_Core_SelectValues::membershipTokens(), $tokens);
-    }
-    elseif ($className == 'CRM_Event_Form_ManageEvent_ScheduleReminders') {
-      $tokens = array_merge(CRM_Core_SelectValues::eventTokens(), $tokens);
-    }
-
```
Comment (davecivicrm) [inline CRM/Mailing/BAO/Mailing.php]: Tim - It looks like the old behavior was to expose ONLY eventTokens to ScheduleReminders form IF it is called from the Manage Events UI. Your refactor seems to expose activity, event and membership tokens for any instance of the ScheduleReminder form. Am I reading this correctly? Did you test what happens if use selects a membership token for an event-related reminder?

Comment (mallezie) [inline CRM/Mailing/BAO/Mailing.php]: Thanks for the review @davecivicrm 
When checking the current behavior on master it seems all tokens (activity / membership / event) are added to the schedule reminder form regardless if it's called from the event UI or from the admin/communication/scheduled reminders UI. 
So this seems to keep the current behaviour. 
If i see the current code correctly the CRM_Event_Form_ManageEvent_ScheduleReminders form adds indeed only event tokens, however this form doesn't use tokens. 
The git log tells this form was cleaned up in https://issues.civicrm.org/jira/browse/CRM-15291 and https://github.com/civicrm/civicrm-core/pull/4644 and it seems this code was a leftover which could be removed there.

File: CRM/Contribute/Form/Task/PDFLetter.php [hunk 2]
```diff
@@ -167,4 +167,14 @@ public function postProcess() {
     CRM_Contribute_Form_Task_PDFLetterCommon::postProcess($this);
   }
 
+  /**
+   * List available tokens for this form.
+   *
+   * @return array
+   */
+  public function listTokens() {
+    $tokens = CRM_Core_SelectValues::contactTokens();
```
Comment (davecivicrm) [inline CRM/Contribute/Form/Task/PDFLetter.php]: This should have also merged in contributionTokens. Fixing this in https://issues.civicrm.org/jira/browse/CRM-17580

**Pull request merged**